0065-drop-dunder-utils.md

```diff
@@ -85,20 +101,48 @@ There are some things which we just can't control, see:
 ## Unresolved questions
 [unresolved]: #unresolved-questions
 
-None so far.
+### Many others may also be using ``salt://_utils/``. That can't be replaced with a simple import. That code would need to be packaged and installed separately into the minion's environment. — By [James Howe](https://github.com/OrangeDog)
+
+What exactly would be a use case for the ``salt://_utils/`` usage?
+
 
 ## Resolved questions
 [resolved]: #resolved-questions
 
-### Is this loader accessible anywhere else? e.g. via utils in templates? That significantly increases the surface of the breaking change. But that's not necessarily a bad thing.
+### Is this loader accessible anywhere else? e.g. via utils in templates? That significantly increases the surface of the breaking change. But that's not necessarily a bad thing. — By [James Howe](https://github.com/OrangeDog)
 
 Thank god no.
 
-### Breaking change on public API for custom modules (and extension modules?)
+### Breaking change on public API for custom modules (and extension modules?) — By [James Howe](https://github.com/OrangeDog)
 
 During the deprecation period, ``__utils__`` shall be available, although a warning about it's
 deprecation will be emmitted.
 
+### But still, is there a cleaner way to solve the original use-case (distribute common code, or override built-in utils)? — By [Max Arnold](https://github.com/max-arnold)
+
+Distributing common code has been solved in the python ecosystem, ie, packaged libraries.
+As for overriding built-in utils, this is one of the major reasons why the utils loader should have never been implemented.
+One thing is being able to override an execution module, or state module, the exposure surface is small, as for
+utility code, the surface is unmeasureable because that code is meant to be reused accross the whole code base. One small change
+in the overridden code has the potential of bringing a Salt installation to the ground.
+
+### I guess shipping custom utils through salt extensions will be also broken? — By [Max Arnold](https://github.com/max-arnold)
+
+A salt extension is a packaged python library that Salt can use, so, why not just import the utilitary code and just use it.
+
+### I do not mind to remove ``__utils__``, but what mechanism should I use now? — By [Alberto Planas](https://github.com/aplanas)
+
+Packaged libraries distributed to the minions. Salt can even distribute and install these on the minions.
+
+### For example, there are shared code that I do not want to be represented as a module. — By [Alberto Planas](https://github.com/aplanas)
+
+I don't know enough about this particular usage or it's reasonings, so, why not?
```

Comment:
Because it is a supporting function, with a non stable API. One [example](https://github.com/openSUSE/yomi/blob/master/salt/_utils/disk.py#L31) is when we have one code reused multiple times in different modules and we do not want to replicate code (and is not big enough to have entity to be an RPM package)

Another [example](https://github.com/openSUSE/yomi/blob/master/salt/_utils/lp.py#L47) is when we have some code that makes very complicated simulation, and to abstract it we need to use classes and stuff.

Basically is a bunch of internal code, that is used by the external API when needed as a [normal Python code](https://github.com/openSUSE/yomi/blob/master/salt/_modules/partmod.py#L35) later.

There is no natural way to express this code as a module, unless I create some weird interface with some documentation with a warning of "do not use this". I think that I remember this exact case in some of the salt modules: some function with this kind of warning, that was designed to be used internally by another salt module.

Removing the `__utils__` without a functional alternative will create more of those public functions that should not be used by any user.